src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -378,6 +411,69 @@ internal ValueTask InternalCopyToAsync(Stream stream, TransportContext? context,
             return new ValueTask(task);
         }
 
+        internal void LoadIntoBuffer(long maxBufferSize, CancellationToken cancellationToken)
+        {
+            CheckDisposed();
+
+            if (!CreateTemporaryBuffer(maxBufferSize, out MemoryStream? tempBuffer, out Exception? error))
+            {
+                // If we already buffered the content, just return.
+                return;
+            }
+
+            if (tempBuffer == null)
+            {
+                throw error!;
+            }
+
+            // Register for cancellation and tear down the underlying stream in case of cancellation/timeout.
+            CancellationTokenRegistration cancellationRegistration = cancellationToken.Register(swr =>
+            {
+                var weakThisRef = (WeakReference<HttpContent>)swr!;
+                if (weakThisRef.TryGetTarget(out HttpContent? strongThisRef))
+                {
+                    strongThisRef.Dispose();
+                }
+            }, new WeakReference<HttpContent>(this));
```

Comment:
```suggestion
            CancellationTokenRegistration cancellationRegistration = cancellationToken.Register(s => ((HttpContent)s!).Dispose(), this);
```
We don't need to use a weak reference here.  You saw it used in other cases where we didn't want the CancellationToken to artificially root the state being passed in... but here this is a synchronous method, and `this` is already being rooted by the stack.

Reply:
Also, I think it'd be worth a comment that we're only comfortable disposing of the HttpContent instance like this because LoadIntoBuffer is internal and we'll only be using it on content instances we get back from a handler's SendAsync call and that haven't been given out to the user yet.  If we were to ever make LoadIntoBuffer public, we'd need to rethink this.